src/mc-fle2-find-text-payload.c

```diff
@@ -0,0 +1,477 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <bson/bson.h>
+#include <stdlib.h>
+
+#include "mc-fle2-find-text-payload-private.h"
+#include "mc-parse-utils-private.h"
+#include "mongocrypt-buffer-private.h"
+#include "mongocrypt-util-private.h"
+#include "mongocrypt.h"
+
+#define IF_FIELD(Name)                                                                                                 \
+    if (0 == strcmp(field, #Name)) {                                                                                   \
+        if (has_##Name) {                                                                                              \
+            CLIENT_ERR("Error parsing %s: Duplicate field '" #Name "'", class_name);                                   \
+            goto fail;                                                                                                 \
+        }                                                                                                              \
+        has_##Name = true;
+
+#define END_IF_FIELD                                                                                                   \
+    continue;                                                                                                          \
+    }
+
+#define CHECK_HAS(Name)                                                                                                \
+    if (!has_##Name) {                                                                                                 \
+        CLIENT_ERR("Error parsing %s: Missing required field '" #Name "'", class_name);                                \
+        goto fail;                                                                                                     \
+    }
+
+#define PARSE_BINARY(Name, Dest)                                                                                       \
+    IF_FIELD(Name) {                                                                                                   \
+        if (!parse_bindata(BSON_SUBTYPE_BINARY, &iter, Dest, status)) {                                                \
+            goto fail;                                                                                                 \
+        }                                                                                                              \
+    }                                                                                                                  \
+    END_IF_FIELD
+
+typedef struct {
+    _mongocrypt_buffer_t *edcDerivedToken;
+    _mongocrypt_buffer_t *escDerivedToken;
+    _mongocrypt_buffer_t *serverDerivedToken;
+} mc_TextFindTokenSetIndirection_t;
+
+typedef struct {
```

Comment:
Why do we need this? Can't we just use `const mc_TextFindTokenSetIndirection_t`?

Reply:
I see, thanks for the explanation.